source/appModules/devenv.py

```diff
@@ -503,3 +571,95 @@ def _get_positionInfo(self):
 		return {
 			"level": int(self.IAccessibleObject.accValue(self.IAccessibleChildID))
 		}
+
+class IntellisenseMenuItem(UIA.UIA):
+
+	def _get_highlighted(self):
+		itemStatus = None
+
+		try:
+			itemStatus = self.UIAElement.CachedItemStatus
+		except COMError:
+			try:
+				itemStatus = self.UIAElement.CurrentItemStatus
+			except:
+				# Item is not initialized yet, we can safely ignore this
+				pass
+
+		# Return false, we can't obtain the status
+		if not itemStatus:
+			return False
+
+		return "[HIGHLIGHTED]=True" in itemStatus
+
+	def event_UIA_itemStatus(self):
+		editor = api.getFocusObject()
+
+		if not self.appModule.messageRead:
+			# See #9739's edit note.
+			self.appModule.messageRead = True
+
+			# See below
+			self.appModule.speechStopped = True
+			speech.cancelSpeech()
+
+			if isinstance(editor, CodeEditor):
+				editor.event_suggestionsOpened()
+			try:
+				ui.message(self.parent.next.next.name)
+			except AttributeError:
+				# This happen when you open and press escape quickly.
+				pass
+
+		if self.highlighted:
+			if self.appModule.intellisenseItem != self:
+				if not self.appModule.speechStopped:
+					# Sometimes the message is read, but speech is stopped
+					# again, so we must use this flag to ensure the help text
+					# is read before the item
+					speech.cancelSpeech()
+
+				if (not self.appModule.intellisenseItem 
+					and not self.appModule.messageRead):
+					# In some project editors the "highlighted" flag is
+					# correctly set to true only when an item is selected,
+					# but in others this is not the case.
+					self.appModule.messageRead = True
+
+					# Same here, fire suggestionsOpened and speak help text
+					if isinstance(editor, CodeEditor):
+						editor.event_suggestionsOpened()
+
+					try:
+						ui.message(self.parent.next.next.name)
+					except AttributeError:
+						pass
+
+				api.setNavigatorObject(self)
+				# We are only interested in name and role properties.

```

Comment:
Would it not be useful to also report description? then arrowing up and down will automatically report the descriptions? For example, the location and signature etc?

Reply:
@michaelDCurran yes, but sometimes this kind of information is not in the description, but in another pop-up.
I omit these just for consistency..

This is related to #9777. When an item is selected a pop-up with documentation is shown.

_Sent from my Galaxy J7 Prime using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_